library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/service/GafferPopNamedOperationServiceIT.java

```diff
@@ -0,0 +1,186 @@
+        GafferPopGraph gafferPopGraph = GafferPopGraph.open(TEST_CONFIGURATION, getGafferGraph());
+        GraphTraversalSource g = gafferPopGraph.traversal();
+
+        // When
+        Map<String, String> params = Collections.singletonMap("execute", "missingNamedOp");
+
+        // Then
+        assertThatThrownBy(() -> g.call("namedoperation", params).toList())
+                .isExactlyInstanceOf(IllegalStateException.class)
+                .hasMessage("Named Operation not found");
+    }
+
+    @Test
+    void shouldAddNamedOperation() {
+        // Given
+        final GafferPopGraph gafferPopGraph = GafferPopGraph.open(TEST_CONFIGURATION, getGafferGraph());
+        GraphTraversalSource g = gafferPopGraph.traversal();
+
+        // When
+        Map<String, Object> addParams = new HashMap<>();
+        addParams.put("name", "testNamedOp");
+        addParams.put("opChain", getOpChainElementCount(null).getOperationChainAsString());
+        Map<String, Object> params = Collections.singletonMap("add", addParams);
+
+        // Then
+        assertThatNoException().isThrownBy(() -> g.call("namedoperation", params).toList());
+    }
+
+    @Test
+    void shouldExecuteNonIterableReturningNamedOperation() throws OperationException {
+        // Given
+        String opName = "testNamedOpNonIterableReturn";
+        Graph gafferGraph = getGafferGraph();
+        gafferGraph.execute(getOpChainElementCount(opName), new User(USER_ID));
+        GafferPopGraph gafferPopGraph = GafferPopGraph.open(TEST_CONFIGURATION, gafferGraph);
+        GraphTraversalSource g = gafferPopGraph.traversal();
+
+        // When
+        Map<String, String> params = Collections.singletonMap("execute", opName);
+
+        // Then
+        assertThat(g.call("namedoperation", params).next())
+                .isInstanceOf(Long.class)
+                .isEqualTo(0L);
+    }
+
+    @Test
+    void shouldExecuteIterableReturningNamedOperation() throws OperationException {
+        // Given
+        String opName = "testNamedOpIterableReturn";
+        Graph gafferGraph = getGafferGraph();
+        gafferGraph.execute(getOpChainElements(opName), new User(USER_ID));
+        GafferPopGraph gafferPopGraph = GafferPopGraph.open(TEST_CONFIGURATION, gafferGraph);
+        GraphTraversalSource g = gafferPopGraph.traversal();
+
+        // When
+        Map<String, String> params = Collections.singletonMap("execute", opName);
+
+        // Then
+        assertThat(g.call("namedoperation", params).toList())
+                .isEmpty();
+    }
+
+    private Graph getGafferGraph() {
+        return new Graph.Builder()
+                .config(new GraphConfig.Builder()
+                        .graphId("graph1")
+                        .build())
+                .storeProperties(PROPERTIES)
+                .addSchemas(StreamUtil.openStreams(this.getClass(), "/gaffer/schema"))
+                .build();
+    }
+
+    private AddNamedOperation getOpChainElementCount(String name) {
```

Comment:
This could maybe be called something like `getAddNamedOpElementCount`

Reply:
I think previously I was using an op chain, but there was no way to directly get the JSON representation of the chain with that. So I switched to the full operation, but forgot to change the method name. I'll fix this.